credsweeper/credentials/candidate.py

```diff
@@ -37,18 +36,16 @@ def __init__(self,
         self.rule_name = rule_name
         self.severity = severity
         self.config = config
-        self.use_ml = use_ml
+        # None - ML is applicable but not processed yet; "NA" - ML is not applicable; float - the ml decision
+        # Note: -1.0 is possible too for some activation functions in ml model, so let avoid negative values
+        self.ml_probability: Union[None, str, float] = None if use_ml else "NA"
```

Comment:
What about use `None` not `"NA"` for `ml_probability`?
I think `"NA"` seems better to convey the meaning.